src/app-layer-parser.c

```diff
@@ -56,7 +56,7 @@
 #include "app-layer-imap.h"
 
 struct AppLayerParserThreadCtx_ {
-    void *alproto_local_storage[FLOW_PROTO_MAX][ALPROTO_MAX];
+    void *(*alproto_local_storage)[FLOW_PROTO_MAX];
```

Comment:
We had a fixed-size 2-dimensional array of pointers (void *)

Now we have a fixed-size 1-dimensional array of pointers-as-pointer-to-an-array of pointers (void *)

This allows to keep notation `alproto_local_storage[alproto][flow_proto]` (as we know one dimension)

Alternative would be to use `void **alproto_local_storage` and index/access it like `alproto_local_storage[alproto*FLOW_PROTO_MAX+flow_proto]` (or have many allocations instead of just one contiguous `SCCalloc(ALPROTO_MAX*FLOW_PROTO_MAX, sizeof(void *))`)